[11][ADD] connector_importer

simahawk opened the pull request:
**TODO**
- [x] add README
- [x] more tests
- [x] fixes and cleanup

simahawk committed:
[ADD] connector_importer

simahawk committed:
[11.0] connector_importer

simahawk committed:
more tests and cleanup

simahawk committed:
Update oca dependencies

mpanarin commented on connector_importer/models/import_type.py:
```diff
@@ -0,0 +1,67 @@
+    The importer machinery will run the imports for all the models declared
+    and will retrieve their specific importerts to execute them.
+    """
+    _name = 'import.type'
```
keep the naming`importer.type`?

mpanarin commented on connector_importer/components/importer.py:
```diff
@@ -0,0 +1,355 @@
+    See `run` method for detailed information on what it does.
+    """
+
+    _name = 'importer.record'
```
imo, this is an extremely confusing name with `import.record`. Maybe it's better to follow general naming and call it `importer.importer` or smth

simahawk replied:
this is the importer of the record.
The distinction overall, is this:
`import.` identifies odoo records
`importer.` identifies what has to do w/ importers (importers, mappers, etc)

simahawk replied:
worth to mention: I spent a bit of time in finding the right way to name those things... I never came up w/ a good solution w/out messing up w/ extra super-long names :cry: 
But I'm open to suggestions :wink:

mpanarin commented on connector_importer/models/record.py:
```diff
@@ -0,0 +1,115 @@
+    When the importer will run, it will read all the records,
+    convert them using connector mappers and do the import.
+    """
+    _name = 'import.record'
```
well, same

mpanarin commented on connector_importer/tests/test_recordset_importer.py:
```diff
@@ -0,0 +1,100 @@
+class TestRecordsetImporter(TestImporterBase):
+
+    @mock.patch('%s.run_import' % RECORD_MODEL)
+    def test_recordset_importer(self, mocked_run_inport):
```
s/inport/import

mpanarin reviewed (approved):
i'd keep naming same across models to prevent confusion
other than a few naming suggestions LGTM

simahawk committed:
connector_importer: add xmlid converter

simahawk committed:
connector_importer: load mapper options on each chunk, log values debug

simahawk committed:
c_importer: use serialized fields, add shared storage

simahawk committed:
fix tests

simahawk committed:
make record handler and tracker real components

mpanarin committed:
[FIX] invalidate_cache for set_shared (#3)

* [FIX] invalidate_cache for set_shared

* fixup! [FIX] invalidate_cache for set_shared

Commit:
Correct method name in autogenerated cron

simahawk committed:
Preserve original values on create/write

simahawk committed:
Record handler: allow force of create_uid/write_uid

simahawk committed:
Record handler: allow force of create_date

Commit:
[FIX] check None instead of falsy on reporter

Commit:
[FIX] base64 conversion

Commit:
[FIX] for csv writer as it expects a string in python3

simahawk committed:
Merge pull request #4 from mpanarin/11-add-connector_importer

simahawk committed:
Recordset importer: drop existing records before run

simahawk committed:
Record handler better handling of create/write ctx

The odoorecord handler now takes a custom context, if any,
from the importer. In this way you can customize it via importer
w/out having to define a custom record handler for each importer.

All create/write actions now pass a new ctx key around:

    `connector_importer_session`

Additionally, `tracking_disable` is ENABLED by default
since it's very unlikely that you want to track all the changes
when import from external sources.

TODO: make tracking disabling customizeable via recordset conf.

simahawk committed:
Record: store JSON data as attachment

simahawk committed:
Add `is_last_importer` flag to mark last importer for a recorset

simahawk committed:
Add FIXME reminder for job relations

simahawk committed:
PEP8

mpanarin commented on connector_importer/models/record.py:
```diff
@@ -84,12 +92,12 @@ def debug_mode(self):
     @api.multi
     @job
-    def import_record(self, component_name, model_name):
+    def import_record(self, component_name, model_name, is_last_importer=True):
```
I think it's worth leaving the small note on what does parameter `id_last_importer` does. As don't find it self-explanatory

simahawk replied:
done

mpanarin commented:
old tests should be updated to check the tuple of three instead of 2

simahawk commented:
@mpanarin fixed tnx

Commit:
[IMP] always create records, when unique key is not defined

Commit:
[FIX] csv file conversion

Commit:
[FIX] report generation

simahawk committed:
Merge pull request #6 from mpanarin/11-imp-connector_importer

11 imp connector importer

Commit:
[FIX] filesize computation

simahawk committed:
Merge pull request #7 from mpanarin/11-imp-connector_importer

[FIX] filesize computation

gurneyalex committed:
update README

simahawk committed:
Merge pull request #8 from gurneyalex/pr44

update README

gurneyalex commented:
/ocabot merge

gurneyalex committed:
connector_importer: disable some lint checks (#9)

gurneyalex commented:
/ocabot merge
